### PR TITLE
feat: Added support for clicked button data for Cognite3DViewer.on('click',...) method

### PR DESCRIPTION
**Pull request by Savokr**
Support for clicked button data for external click handling using Cognite3DViewer. Also includes small refactoring of EventDelegates so that they are in one place because before there were 3 exactly the same version of PointerEventDelegate in different places. Now all events related types are located in utilities/events and can be imported from there by any internal package.

### CODE REVIEW

File: examples/src/pages/Geomap.tsx [hunk 1]
```diff
@@ -177,7 +177,7 @@ export function Geomap() {
       const selectedSet = new TreeIndexNodeCollection([]);
 
       new AxisViewTool(viewer);
-      viewer.on('click', async event => {
+      viewer.on('click', async (event: { offsetX: any; offsetY: any; }) => {
```
Comment (christjt) [inline examples/src/pages/Geomap.tsx:180]: remove this

File: viewer/core/src/public/migration/types.ts [hunk 2]
```diff
@@ -204,31 +204,26 @@ export { CameraConfiguration } from '@reveal/utilities';
...
  * @module @cognite/reveal
  * @see {@link Cognite3DViewer.on}.
  */
-export type SceneRenderedDelegate = (event: {
-  frameNumber: number;
-  renderTime: number;
-  renderer: THREE.WebGLRenderer;
-  camera: THREE.PerspectiveCamera;
-}) => void;
+export { SceneRenderedDelegate } from '@reveal/utilities';
```
Comment (christjt) [inline viewer/core/src/public/migration/types.ts:226]: Still don't understand why the core package is exporting things from utilities.

File: viewer/packages/camera-manager/src/CameraManager.ts [hunk 3]
```diff
@@ -2,7 +2,8 @@
  * Copyright 2022 Cognite AS
  */
 
-import { CameraChangeDelegate, CameraState } from './types';
+import { CameraState } from './types';
+import { CameraChangeDelegate } from '@reveal/utilities';
```
Comment (christjt) [inline viewer/packages/camera-manager/src/CameraManager.ts:6]: It seems very strange to me that the camera change delegate is a part of utilities.

File: viewer/packages/utilities/src/events/types.ts [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+
+/**
+ * Delegate for pointer events.
+ */
+export type PointerEventDelegate = (event: { offsetX: number; offsetY: number; button?: number }) => void;
+
+/**
+ * Delegate for camera update events.
+ */
+export type CameraChangeDelegate = (position: THREE.Vector3, target: THREE.Vector3) => void;
```
Comment (christjt) [inline viewer/packages/utilities/src/events/types.ts:13]: I feel like this belongs in the camera package. I am open for arguments though.

**Review by christjt — CHANGES_REQUESTED** [review]
Overall looks good. Still not convinced that the `CameraChangeDelegate` should live in utilities.

File: viewer/core/src/public/migration/types.ts (hunk 2 above)
Comment (Savokr) [reply viewer/core/src/public/migration/types.ts:226]: Now exported from top level in viewer

File: viewer/packages/camera-manager/src/CameraManager.ts (hunk 3 above)
Comment (Savokr) [reply viewer/packages/camera-manager/src/CameraManager.ts:6]: Move to CameraManager package

**Pull request merged**